internal/service/advancedclustertpf/model_ClusterDescription20240805.go

```diff
@@ -25,13 +25,18 @@ type LegacySchemaInfo struct {
 	RootDiskSize               *float64
 }
 
-func NewTFModel(ctx context.Context, input *admin.ClusterDescription20240805, timeout timeouts.Value, diags *diag.Diagnostics, legacyInfo *LegacySchemaInfo) *TFModel {
+type ExtraAPIInfo struct {
+	ContainerIDs map[string]string
+}
+
+func NewTFModel(ctx context.Context, input *admin.ClusterDescription20240805, timeout timeouts.Value, diags *diag.Diagnostics, legacyInfo *LegacySchemaInfo, apiInfo *ExtraAPIInfo) *TFModel {
 	biConnector := NewBiConnectorConfigObjType(ctx, input.BiConnector, diags)
 	connectionStrings := NewConnectionStringsObjType(ctx, input.ConnectionStrings, diags)
 	labels := NewLabelsObjType(ctx, input.Labels, diags)
-	replicationSpecs := NewReplicationSpecsObjType(ctx, input.ReplicationSpecs, diags, legacyInfo)
+	replicationSpecs := NewReplicationSpecsObjType(ctx, input.ReplicationSpecs, diags, legacyInfo, apiInfo)
 	tags := NewTagsObjType(ctx, input.Tags, diags)
-	if diags.HasError() {
+	if diags.HasError() || apiInfo == nil {
```

Comment:
why are we checking apiInfo == null when we used apiInfo earlier in NewReplicationSpecsObjType, shouldn't that func add errors to diag in case there is some problem including apiInfo == null?

Reply:
Good point. Changing